--- a/menu/cbs/menu_cbs_get_value.c	2023-07-01 23:53:04.783174800 +0200
+++ b/menu/cbs/menu_cbs_get_value.c	2023-07-01 23:55:00.817479760 +0200
@@ -655,7 +655,7 @@
       const char *path,
       char *s2, size_t len2)
 {
-   size_t _len;
+
    unsigned policyid              = atoi(path);
    cpu_scaling_driver_t **drivers = get_cpu_scaling_drivers(false);
    cpu_scaling_driver_t *d        = drivers[policyid];
