--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -340,6 +340,9 @@
 if(ARM64)
 	message("Generating for ARMv8, ${CMAKE_BUILD_TYPE}")
 endif()
+
+# Temp workaround for https://github.com/hrydgard/ppsspp/issues/18688
+add_compile_options(-flax-vector-conversions)
 
 # It looks like the flags for the selected build type are written to the cache after each run, which causes some of the operations
 # below to keep expanding them with the same flags over and over on every run, leading to a rebuild of the majority of the files.

