diff -u /home/romain/mame/src/mame/fujitsu/fmtowns.cpp ./src/mame/fujitsu/fmtowns.cpp 
--- /home/romain/mae/src/mame/fujitsu/fmtowns.cpp	2023-08-09 10:51:01.471695506 +0200
+++ ./src/mame/fujitsu/fmtowns.cpp	2023-08-09 10:47:03.165680100 +0200
@@ -905,7 +905,7 @@
 	// Tatsujin Ou expects it to read as zero to navigate menus.
 	// Unclear whether it always reads as zero, or it's affected by something undocumented.
 	unsigned const pad = BIT(offset, 1);
-	return m_pad_ports[pad]->read() & (0x8f | (bitswap<3>(m_towns_pad_mask, pad + 4, (pad * 2) + 1, pad * 2) << 4));
+	return m_pad_ports[pad]->read() & (0x0f | (bitswap<3>(m_towns_pad_mask, pad + 4, (pad * 2) + 1, pad * 2) << 4));
 }
 
 void towns_state::towns_pad_mask_w(uint8_t data)
