diff --git a/src/video.cpp b/src/video.cpp
index 876e0f5..7adfd58 100755
--- a/src/video.cpp
+++ b/src/video.cpp
@@ -727,7 +727,7 @@ inline void prepareScreen(int width, int height)
         y = 0;
         h = go2_display_height_get(display);
         w = go2_display_width_get(display);
-        isTate = (Retrorun_Core == RETRORUN_CORE_FLYCAST); // we rotate the screen (Tate) for some arcade games when apsect ratio < 0
+        //isTate = (Retrorun_Core == RETRORUN_CORE_FLYCAST); // we rotate the screen (Tate) for some arcade games when apsect ratio < 0
     }
 }
 
@@ -1073,4 +1073,4 @@ void core_video_refresh(const void *data, unsigned width, unsigned height, size_
                            x, y, w, h,
                            _351Rotation);
     }
-}
\ No newline at end of file
+}
