diff --git a/src/input.cpp b/src/input.cpp
index 0f2e945..1a9a2d2 100755
--- a/src/input.cpp
+++ b/src/input.cpp
@@ -395,13 +395,13 @@ int16_t core_input_state(unsigned port, unsigned device, unsigned index, unsigne
         const float TRIM = 0.35f;
         go2_thumb_t thumb = go2_input_state_thumbstick_get(gamepadState, Go2InputThumbstick_Left);
 
-        if (thumb.y < -TRIM)
+        if (thumb.y < TRIM)
             go2_input_state_button_set(gamepadState, Go2InputButton_DPadUp, ButtonState_Pressed);
-        if (thumb.y > TRIM)
+        if (thumb.y > -TRIM)
             go2_input_state_button_set(gamepadState, Go2InputButton_DPadDown, ButtonState_Pressed);
-        if (thumb.x < -TRIM)
+        if (thumb.x < TRIM)
             go2_input_state_button_set(gamepadState, Go2InputButton_DPadLeft, ButtonState_Pressed);
-        if (thumb.x > TRIM)
+        if (thumb.x > -TRIM)
             go2_input_state_button_set(gamepadState, Go2InputButton_DPadRight, ButtonState_Pressed);
     }
 